pccommon/pccommon/tables.py

```diff
@@ -0,0 +1,206 @@
+        self._table_client: Optional[TableClient] = None
+        self._cache: Cache = TTLCache(maxsize=1024, ttl=ttl or DEFAULT_TABLE_TTL)
+
+    def _ensure_table_client(self) -> None:
+        if not self._table_client:
+            raise TableError("Table client not initialized. Use as a context manager.")
+
+    def __enter__(self) -> TableClient:
+        self._service_client, self._table_client = self._get_clients()
+        return self._table_client
+
+    def __exit__(self, *args: Any) -> None:
+        if self._table_client:
+            self._table_client.close()
+            self._table_client = None
+        if self._service_client:
+            self._service_client.close()
+            self._service_client = None
+
+    @classmethod
+    def from_sas_token(
+        cls: Type[T], account_url: str, sas_token: str, table_name: str
+    ) -> T:
+        def _get_clients(
+            _url: str = account_url, _token: str = sas_token, _table: str = table_name
+        ) -> Tuple[Optional[TableServiceClient], TableClient]:
+            table_service_client = TableServiceClient(
+                endpoint=_url,
+                credential=AzureSasCredential(_token),
+            )
+            return (
+                table_service_client,
+                table_service_client.get_table_client(table_name=_table),
+            )
+
+        return cls(_get_clients)
+
+    @classmethod
+    def from_connection_string(
+        cls: Type[T], connection_string: str, table_name: str
+    ) -> T:
+        def _get_clients(
+            _conn_str: str = connection_string, _table: str = table_name
+        ) -> Tuple[Optional[TableServiceClient], TableClient]:
+            table_service_client = TableServiceClient.from_connection_string(
+                conn_str=_conn_str
+            )
+            return (
+                table_service_client,
+                table_service_client.get_table_client(table_name=_table),
+            )
+
+        return cls(_get_clients)
+
+    @classmethod
+    def from_account_key(
+        cls: Type[T],
+        account_name: str,
+        account_key: str,
+        table_name: str,
+        account_url: Optional[str] = None,
+        ttl: Optional[int] = None,
+    ) -> T:
+        def _get_clients(
+            _name: str = account_name,
+            _key: str = account_key,
+            _url: Optional[str] = account_url,
+            _table: str = table_name,
+        ) -> Tuple[Optional[TableServiceClient], TableClient]:
+            _url = _url or f"https://{_name}.table.core.windows.net"
+            credential = AzureNamedKeyCredential(name=_name, key=_key)
+            table_service_client = TableServiceClient(
+                endpoint=_url, credential=credential
+            )
+            return (
+                table_service_client,
+                table_service_client.get_table_client(table_name=_table),
+            )
+
+        return cls(_get_clients, ttl=ttl)
```

Comment:
When trying to load a tile url, I'm getting an exception here:

```
tiler_1     |   File "/opt/src/pccommon/pccommon/config/core.py", line 53, in get_container_config_table
tiler_1     |     return ContainerConfigTable.from_account_key(
tiler_1     |   File "/opt/src/pccommon/pccommon/tables.py", line 132, in from_account_key
tiler_1     |     return cls(_get_clients, ttl=ttl)
tiler_1     | TypeError: __init__() got an unexpected keyword argument 'ttl'
```

You should be able to reproduce it with a local url like:

http://localhost:8080/data/item/preview.png?collection=chloris-biomass&item=chloris_biomass_50km_2019&assets=biomass&colormap_name=chloris-biomass&rescale=1,750000&max_size=100

Reply:
Fixed in [e090d66](https://github.com/microsoft/planetary-computer-apis/pull/48/commits/e090d667eed52b74815cc45052287b3afb5ce9e8)